libs/language-server/src/stdlib/domain/mobility/GTFSValueTypes.jv

```diff
@@ -0,0 +1,99 @@
+ * Float - A floating point number.
+ * Integer - An integer.
+ * Phone number - A phone number.
+ * Time - Time in the HH:MM:SS format (H:MM:SS is also accepted). The time is measured from "noon minus 12h" of the service day (effectively midnight except for days on which daylight savings time changes occur). For times occurring after midnight on the service day, enter the time as a value greater than 24:00:00 in HH:MM:SS.
+ *      Example: 14:30:00 for 2:30PM or 25:35:00 for 1:35AM on the next day.
+ * Text - A string of UTF-8 characters, which is aimed to be displayed and which must therefore be human readable.
+ * Timezone - TZ timezone from the https://www.iana.org/time-zones. Timezone names never contain the space character but may contain an underscore. Refer to http://en.wikipedia.org/wiki/List_of_tz_zones for a list of valid values.
+ *      Example: Asia/Tokyo, America/Los_Angeles or Africa/Cairo.
+ * URL - A fully qualified URL that includes http:// or https://, and any special characters in the URL must be correctly escaped. See the following http://www.w3.org/Addressing/URL/4_URI_Recommentations.html for a description of how to create fully qualified URL values.
+ * 
+ * https://gtfs.org/schedule/reference/#field-types
+*/
+
+constraint GTFSColorConstraint on text: value matches /^[A-F]{6}$/;
+valuetype GTFSColor oftype text {
+    constraints: [GTFSColorConstraint];
+}
+
+constraint DateYYYYMMDD on text: value matches /^[0-9]{4}[0-9]{2}[0-9]{2}$/;
+valuetype GTFSDate oftype text {
+    constraints: [DateYYYYMMDD];
+}
+
+constraint CurrencyConstraint on text: value matches /^[A-Z]{3}$/;
+valuetype GTFSCurrency oftype text {
+    constraints: [CurrencyConstraint];
+}
+
+constraint TimeHHMMSS on text: value matches /^[0-9]{1,2}:{1}[0-9]{2}:{1}[0-9]{2}$/;
+valuetype GTFSTime oftype text {
+    constraints: [TimeHHMMSS];
+}
+
+// Placeholders
+constraint EnumTwo on integer: value in [0, 1];
+constraint EnumOneOrTwo on integer: value in [1, 2];
+constraint EnumThree on integer: value in [0, 1, 2];
+
+valuetype GTFSEnumTwo oftype integer {
+    constraints: [EnumTwo];
+}
+
+valuetype GTFSEnumOneOrTwo oftype integer {
+    constraints: [EnumOneOrTwo];
+}
+
+valuetype GTFSEnumThree oftype integer {
+    constraints: [EnumThree];
+}
+
+// Generic value types
+constraint Latitude on decimal: value >= -90 and value <= 90;
+valuetype GTFSLatitude oftype decimal {
+    constraints: [Latitude];
+}
+
+constraint Longitude on decimal: value >= -180 and value <= 180;
+valuetype GTFSLongitude oftype decimal {
+    constraints: [Longitude];
+}
+
+constraint NonNegativeNumber on decimal: value >= 0;
+valuetype GTFSNonNegativeDecimal oftype decimal {
+    constraints: [NonNegativeNumber];
+}
+
+valuetype GTFSNonNegativeInteger oftype integer {
+    constraints: [NonNegativeNumber];
+}
+
+constraint URLIncludingSchema on text: value matches /^(http)s?(:\/\/)/;
+valuetype GTFSUrl oftype text {
+    constraints: [URLIncludingSchema];
+}
```

Comment:
Add a new line
> “lines” not ending in a newline character aren't considered actual lines.

(https://stackoverflow.com/questions/729692/why-should-text-files-end-with-a-newline)

Reply:
Wow I've learned something today, thanks.